.github/workflows/ubuntu.yml

```
@@ -35,10 +35,13 @@ jobs:
                 -DNGX_OTEL_DEV=ON ..
           make -j $(nproc)
       - name: Download otelcol
+        env:
+          URL: "https://github.com/open-telemetry/\
+            opentelemetry-collector-releases/releases/latest/"
         run: |
-          curl -sLo - https://github.com/\
-          open-telemetry/opentelemetry-collector-releases/releases/download/\
-          v0.115.1/otelcol_0.115.1_linux_amd64.tar.gz | tar -xzv
+          curl -sLo - ${URL}download/otelcol_\
+          $(curl -sD - $URL | grep location: | cut -d v -f 2 | sed -e 's/\r//')\
+          _linux_amd64.tar.gz | tar -xzv
```

Using Github's API and jq would result in more readable and reliable code:
```suggestion
          LATEST=open-telemetry/opentelemetry-collector-releases/releases/latest
          TAG=$(curl -s https://api.github.com/repos/${LATEST} |
                jq -r .tag_name)
          curl -sLo - https://github.com/${LATEST}/download/\
          otelcol_${TAG:1}_linux_amd64.tar.gz | tar -xzv
```